swift/Workflow/Tests/SubtreeManagerTests.swift

```diff
@@ -294,6 +296,71 @@ final class SubtreeManagerTests: XCTestCase {
         observer.send(value: ())
         wait(for: [notEmittedExpectation], timeout: 1.0)
     }
+
+    // MARK: - SideEffect
+
+    func test_maintainsSideEffectLifetimeBetweenRenderPasses() {
+        let manager = WorkflowNode<ParentWorkflow>.SubtreeManager()
+        XCTAssertTrue(manager.sideEffectLifetimes.isEmpty)
+
+        _ = manager.render { context -> TestViewModel in
+            context.runSideEffect(key: "helloWorld") { _ in }
+            return context.render(
+                workflow: TestWorkflow(),
+                key: "",
+                outputMap: { _ in AnyWorkflowAction.noAction }
+            )
+        }
+
+        XCTAssertEqual(manager.sideEffectLifetimes.count, 1)
+        let sideEffectKey = manager.sideEffectLifetimes.values.first!
+
+        _ = manager.render { context -> TestViewModel in
+            context.runSideEffect(key: "helloWorld") { _ in }
+            return context.render(
+                workflow: TestWorkflow(),
+                key: "",
+                outputMap: { _ in AnyWorkflowAction.noAction }
+            )
+        }
+
+        XCTAssertEqual(manager.sideEffectLifetimes.count, 1)
+        XCTAssertTrue(manager.sideEffectLifetimes.values.first === sideEffectKey)
+    }
+
+    func test_endsUnusedSideEffectLifetimeAfterRenderPasses() {
+        let manager = WorkflowNode<ParentWorkflow>.SubtreeManager()
+        XCTAssertTrue(manager.sideEffectLifetimes.isEmpty)
+
+        let lifetimeEndedExpectation = expectation(description: "Lifetime Ended Expectations")
+        _ = manager.render { context -> TestViewModel in
+            context.runSideEffect(key: "helloWorld") { lifetime in
+                lifetime.onEnded {
+                    // Capturing `lifetime` to make sure a retain-cycle will still trigger the `onEnded` block
+                    print("\(lifetime)")
+                    lifetimeEndedExpectation.fulfill()
+                }
+            }
+            return context.render(
+                workflow: TestWorkflow(),
+                key: "",
+                outputMap: { _ in AnyWorkflowAction.noAction }
+            )
+        }
+
+        XCTAssertEqual(manager.sideEffectLifetimes.count, 1)
+
+        _ = manager.render { context -> TestViewModel in
+            context.render(
+                workflow: TestWorkflow(),
+                key: "",
+                outputMap: { _ in AnyWorkflowAction.noAction }
+            )
+        }
+
+        XCTAssertEqual(manager.sideEffectLifetimes.count, 0)
+        wait(for: [lifetimeEndedExpectation], timeout: 1)
+    }
 }
```

Comment:
Might be redundant to the two tests, but would be nice to have a test with two different keys between two render passes to make sure we perform both of them?